llvm/lib/Transforms/IPO/LowerTypeTests.cpp

```diff
@@ -1392,6 +1402,10 @@ void LowerTypeTestsModule::replaceWeakDeclarationWithJumpTablePtr(
   // Don't use range based loop, because use list will be modified.
   while (!PlaceholderFn->use_empty()) {
     Use &U = *PlaceholderFn->use_begin();
+    if (isFunctionAnnotation(U.getUser())) {
+      U.set(F);
+      continue;
+    }
```

Comment:
Is this check actually needed? If I'm reading the code right, your check inside replaceCfiUses() will already prevent the replacement of F with PlaceholderFn in the annotation, no?

Reply:
Right, this is not needed.  I was originally only considering the function declaration case, and later added code in replaceCfiUses() to cover function definition case.